runtime/service_test.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package main
+
+import (
+	"context"
+	"syscall"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestFindNextAvailableVsockCID(t *testing.T) {
+	sysCall = func(trap, a1, a2, a3 uintptr) (r1, r2 uintptr, err syscall.Errno) {
```

Comment:
I would add the `unix.EADDRINUSE` case and default error case.

`unix.EADDRINUSE` could be squashed into this test
```go
count := 0
sysCall = func(trap, a1, a2, a3 uintptr) (r1, r2 uintptr, err syscall.Errno) {
    if count == 0 {
         return r1, r2, unix.EADDRINUSE
    }
    count++
    return 0, 0, 0
}
```

Reply:
Current implementation is fine. As of now I want to have our code running basic scenarios, not shiny and 100% covered with tests